cloudservice/gcp/gcp.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package gcp
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/DataDog/chaos-controller/cloudservice/types"
+)
+
+type CloudProviderIPRangeManager struct {
+}
+
+// GCPIpRange from the model of the ip range file from GCP
+type GCPIPRange struct {
+	IPPrefix string `json:"ipv4Prefix"`
+	Region   string `json:"scope"`
+	Service  string `json:"service"`
+}
+
+// GCPIpRanges from the model of the ip range file from GCP
+type GCPIPRanges struct {
+	SyncToken string       `json:"syncToken"`
+	Prefixes  []GCPIPRange `json:"prefixes"`
+}
+
+func New() *CloudProviderIPRangeManager {
+	return &CloudProviderIPRangeManager{}
+}
+
+// IsNewVersion Check if the ip ranges pulled are newer than the one we already have
+func (s *CloudProviderIPRangeManager) IsNewVersion(newIPRanges []byte, oldVersion string) bool {
+	ipRanges := GCPIPRanges{}
+	if err := json.Unmarshal(newIPRanges, &ipRanges); err != nil {
```

Comment:
does this take care of missing `newIPRanges`?

Reply:
I'm not sure I'm understanding right what you want to say here. Could you elaborate?

Reply:
😃 sure, i meant to check if `json.Unmarshal([0]byte{}, &ipRanges)` or `json.Unmarshal([0]byte{}, &ipRanges)` could cause problem

Reply:
Added new unit testing cases verifying this